src/config.rs

```diff
@@ -88,23 +138,29 @@ pub struct Manager<'a> {
     env: HashMap<String, String>,
     keyring: Arc<Mutex<Box<dyn credentials::Provider + Send>>>,
     config_file: RwLockWriteGuard<'a, File>,
+    env_user: Option<User>,
+}
+
+enum UserConfigContext {
+    File(usize),
+    Env,
 }
```

Comment:
Thoughts on making this, to me that makes more sense as a type?
```suggestion
enum UserConfigContext {
    File(User),
    Env(User),
}
```

Reply:
It would need to be `&User` rather than `User`. `UserConfigContext` is exclusively for callers of `get_current_user_config_context` (a private method) to consume and decide what they want to do. Since some of them will be in a mutable context (e.g. `set_last_used` from #29) and some won't be (e.g. `get_current_user`), this isn't really an option.